Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -7011,6 +7154,1999 @@ def panorama_install_file_content_update_command(args: dict):
+
+        return ShowRouteSummaryCommandResult(
+            summary_data=summary_data,
+            result_data=[]
+        )
+
+    @staticmethod
+    def get_bgp_peers(topology: Topology,
+                      device_filter_str: str = None) -> ShowRoutingProtocolBGPCommandResult:
+        summary_data = []
+        result_data = []
+        for firewall in topology.firewalls(filter_string=device_filter_str):
+            response = run_op_command(firewall, FirewallCommand.ROUTING_PROTOCOL_BGP_PEER_COMMAND)
+            summary_data.append(dataclass_from_element(firewall, ShowRoutingProtocolBGPPeersSummaryData,
+                                                       response.find("./result/entry")))
+            result_data.append(dataclass_from_element(firewall, ShowRoutingProtocolBGPPeersResultData,
+                                                      response.find("./result/entry")))
+
+        return ShowRoutingProtocolBGPCommandResult(
+            summary_data=summary_data,
+            result_data=result_data
+        )
+
+    @staticmethod
+    def get_device_state(topology: Topology,
+                         device_filter_str: str):
+        for firewall in topology.firewalls(filter_string=device_filter_str):
+            # Connect directly to the firewall
+            direct_firewall_connection: Firewall = topology.get_direct_device(firewall)
+            direct_firewall_connection.xapi.export(category="device-state")
+            return direct_firewall_connection.xapi.export_result.get("content")
+
+    @staticmethod
+    def get_ha_status(topology: Topology,
+                      device_filter_str: str = None) -> list[ShowHAState]:
+
+        result: list[ShowHAState] = []
+        for firewall in topology.all(filter_string=device_filter_str):
+            firewall_host_id: str = resolve_host_id(firewall)
+
+            peer_serial: str = topology.get_peer(firewall_host_id)
+            if not peer_serial:
+                result.append(ShowHAState(
+                    hostid=firewall_host_id,
+                    status="HA Not enabled.",
+                    active=True,
+                    peer=""
+                ))
+            else:
+                state_information_element = run_op_command(firewall, FirewallCommand.HA_STATE_COMMAND)
+                state = find_text_in_element(state_information_element, "./result/group/local-info/state")
+
+                if state == "active":
+                    result.append(ShowHAState(
+                        hostid=firewall_host_id,
+                        status=state,
+                        active=True,
+                        peer=peer_serial
+                    ))
+                else:
+                    result.append(ShowHAState(
+                        hostid=firewall_host_id,
+                        status=state,
+                        active=False,
+                        peer=peer_serial
+                    ))
+
+        if len(result) == 1:
```

Comment:
So this is a good question and it's due to the way XSOAR handles list results. It seems like if you run a command twice, and each time it returns a 1-item list, this will turn into two output values in the context, even if the output_key_field is the same.

So the check here is, if its' a 1 item list, simply return that item, so when it goes into the context we don't create duplicate values.

Reply:
that's interesting, would you mind to take a look at this together with me? lets discuss this further

Reply:
Let me know if you want to jump on a quick call for this one

Reply:
yeah lets jump into a call this week.